apis/v1/secretproviderclass_types.go

```diff
@@ -46,13 +46,32 @@ type SecretObject struct {
 	Data        []*SecretObjectData `json:"data,omitempty"`
 }
 
+type Secret struct {
+	// name of the object to sync
+	ObjectName string `json:"objectName,omitempty"`
+	// expected format of the secret received from the provider (e.g. plaintext, json)
+	Format string `json:"format,omitempty"`
+	// JSON path to target for a secret received in JSON format
+	JsonPath string `json:"jsonPath,omitempty"`
+}
+
+type TransformOptions struct {
+	// expected format of the secret received from the provider (e.g. plaintext, json)
+	Format string `json:"format,omitempty"`
+	// JSON path to target for a secret received in JSON format
+	JsonPath string   `json:"jsonPath,omitempty"`
+	Secrets  []Secret `json:"secrets,omitempty"`
+}
+
 // SecretProviderClassSpec defines the desired state of SecretProviderClass
 type SecretProviderClassSpec struct {
 	// Configuration for provider name
 	Provider Provider `json:"provider,omitempty"`
 	// Configuration for specific provider
 	Parameters    map[string]string `json:"parameters,omitempty"`
 	SecretObjects []*SecretObject   `json:"secretObjects,omitempty"`
+	// Configuration for secret transformation
+	TransformOptions TransformOptions `json:"transformOptions,omitempty"`
```

Comment:
I think we want this to be a list like `[]*TransformOptions` so that multiple transforms can be done on the secrets before presenting in the pod filesystem.